### PR TITLE
Add docs about the testnet

### PR DESCRIPTION
**Pull request by sakridge**

### CODE REVIEW

**Comment by sakridge** [comment]
@mvines @aeyakovenko thoughts?

**Comment by mvines** [comment]
This is great, I should have documented stuff like this already.  Only concern is that a lot of this stuff won't be useable by non-core Solana folks so putting it in a public repo is a little like teasing.

File: doc/testnet.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# TestNet debugging info
+
+Currently we have two testnets, 'perf' and 'master', both on the master branch of the solana repo. Deploys happen at the top of every hour with the latest code. 'perf' has more cores for the client machine to flood
+the network with transactions until failure.
+
+## Deploy process
+
+They are deployed with the `ci/testnet-deploy.sh` script. Validators are selected based on their machine name and everyone gets the binaries installing from snap.
```
Comment (aeyakovenko) [inline doc/testnet.md]: what machine/identity runs this thing?

**Comment by aeyakovenko** [comment]
@sakridge @mvines thats awesome!

i think we need a howto on changing and debugging client code, validator and leader code.

File: doc/testnet.md (hunk 1 above)
Comment (sakridge) [reply doc/testnet.md]: They get run through buildkite, so I think it could be multiple agents that pick up the job. I'll add a sentence about that.

**Pull request merged**